### PR TITLE
Feat/onboarding admin

### PR DESCRIPTION
**Pull request by zomars**
Re-opening #3393 

## What does this PR do?

Implements a wizard to create the first user (admin) as an on-boarding step to the app. The on-boarding will continue to expand in the near future which is why it was implemented this way.

UI:
<kbd>image</kbd>

Can't be used again once at least one user exists in the DB:
<kbd>image</kbd>

Implements #3384 

**Environment**: Staging(main branch) / Production

## Type of change

- [X] New feature (non-breaking change which adds functionality)

## How should this be tested?

Remove all users from the DB and go to the login page (not having a session active). It will show the new onboarding wizard. You can also test going to `/auth/setup` having users in the DB to see the second screen above, indicating the step is done (not needed). The API that supports this new functionality also checks if there is no user in the DB to be able to enable creating an admin user through the new onboarding wizard.

### CODE REVIEW

File: apps/web/pages/auth/setup.tsx [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useRouter } from "next/router";
+import { Controller, useForm } from "react-hook-form";
+import * as z from "zod";
+
+import { isPasswordValid } from "@calcom/lib/auth";
+import classNames from "@calcom/lib/classNames";
+import { useLocale } from "@calcom/lib/hooks/useLocale";
+import { inferSSRProps } from "@calcom/types/inferSSRProps";
+import WizardForm from "@calcom/ui/WizardForm";
```
Comment (zomars) [inline apps/web/pages/auth/setup.tsx:11]: @leog cc @sean-brydon maybe we want to put this in `/v2/` for now. Also, is this the same that we have on console?

File: packages/lib/auth.ts [hunk 2]
```diff
@@ -9,3 +9,19 @@ export async function verifyPassword(password: string, hashedPassword: string) {
   const isValid = await compare(password, hashedPassword);
   return isValid;
 }
+
+export function isPasswordValid(password: string) {
```
Comment (zomars) [inline packages/lib/auth.ts:13]: Let's put this in `packages/lib` instead

File: packages/ui/Stepper.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
+import Link from "next/link";
+
+type DefaultStep = {
+  title: string;
+};
+
+function Stepper<T extends DefaultStep>(props: { href: string; step: number; steps: T[] }) {
```
Comment (zomars) [inline packages/ui/Stepper.tsx:7]: Put it in `/v2/` for now? Also let's try to hurry with #3051

File: packages/ui/WizardForm.tsx [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+import Stepper from "./Stepper";
+
+type DefaultStep = {
+  title: string;
+  description: string;
+  content: JSX.Element;
+  enabled?: boolean;
+};
+
+function WizardForm<T extends DefaultStep>(props: { href: string; steps: T[] }) {
```
Comment (zomars) [inline packages/ui/WizardForm.tsx:12]: Same v2 comment

**Review by zomars — COMMENTED** [review]
Got some Qs. Looking good so far, nice work @leog 🙏🏽

File: apps/web/pages/auth/setup.tsx (hunk 1 above)
Comment (leog) [reply apps/web/pages/auth/setup.tsx:11]: Yes, is the same, moved it from `console` to here and added with a few tweaks. Sure, I can move it to `v2` along the other components.

File: packages/ui/WizardForm.tsx (hunk 4 above)
Comment (sean-brydon) [reply packages/ui/WizardForm.tsx:12]: V2 Should be good to merge into main now - agreed these should live in V2

**Comment by leog** [comment]
Will be reworked once #3051 is merged, converting to draft in the meantime.

File: packages/lib/auth.ts (hunk 2 above)
Comment (leog) [reply packages/lib/auth.ts:13]: @zomars you mean in the ´index.ts´ file instead of ´auth.ts´?

File: apps/web/pages/auth/setup.tsx (hunk 1 above)
Comment (leog) [reply apps/web/pages/auth/setup.tsx:11]: @zomars after chatting with @sean-brydon it seems like we can't move these components to `v2` as there is a color scheme mismatch between that and web where these components are being used. So, for now we should stay put.

File: packages/ui/Stepper.tsx (hunk 3 above)
Comment (leog) [reply packages/ui/Stepper.tsx:7]: We can't move, see comment above.

File: packages/ui/WizardForm.tsx (hunk 4 above)
Comment (leog) [reply packages/ui/WizardForm.tsx:12]: We can't move, see comment above.

File: apps/web/pages/auth/setup.tsx (hunk 1 above)
Comment (zomars) [reply apps/web/pages/auth/setup.tsx:12]: Let's update these components to match the new color scheme. This shouldn't be a blocker

**Comment by leog** [comment]
Here is a Loom with the latest changes: https://www.loom.com/share/ebf85cfb4c4145bf97de5c88b8ccbb6d

**Review by zomars — COMMENTED** [review]
Awesome job @leog I cannot approve my own PR but it is approved in spirit 🙏🏽

**Pull request merged**